MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -1262,10 +1259,10 @@ - (void)saveMetaData
             }
 
             // Store new data
-            [NSKeyedArchiver archiveRootObject:metaData2 toFile:file];
+            [NSKeyedArchiver archiveRootObject:metaData toFile:file];
             
             // Make sure the data will be backed up with the right events stream token from here.
-            backupEventStreamToken = metaData2.eventStreamToken;
+            backupEventStreamToken = metaData.eventStreamToken;
```

Comment:
I would prefer to store the `metaData.eventStreamToken` in `backupEventStreamToken` before archiving the `metaData`, just in case the metaData.eventStreamToken is updated on the main thread during the store saving.